File: concrete/src/main/java/pcb/uwu/unit/base/CelsiusUnit.java
package pcb.uwu.unit.base;

import java.math.BigDecimal;

public class CelsiusUnit extends TemperatureUnit {

	private static final BigDecimal KELVIN_TO_CELSIUS = new BigDecimal("273.15");

	public static final CelsiusUnit CELSIUS = new CelsiusUnit();

	public CelsiusUnit() {
		super("°C", "celsius", "celsius",
				value -> value.minus(KELVIN_TO_CELSIUS),
				value -> value.plus(KELVIN_TO_CELSIUS));
	}
}

File: concrete/src/main/java/pcb/uwu/unit/base/CelsiusUnit.kt
package pcb.uwu.unit.base

import java.math.BigDecimal

object CelsiusUnit
    : TemperatureUnit(symbol = "°C",
                      singularName = "celsius",
                      pluralName = "celsius",
                      translationToCanonical = { it - BigDecimal("273.15") },
                      translationFromCanonical = { it + BigDecimal("273.15") })
